plugins/fileservices/fileservices.cpp

```diff
@@ -198,63 +198,129 @@ static IPropertyTree *getEnvironmentTree(IConstEnvironment * daliEnv)
     return getHPCCEnvironment();
 }
 
-static StringAttr espurl;           // Default ESP url if none specified
+static void setServerAccess(CClientFileSpray &server, IConstWorkUnit * wu)
+{
+    StringBuffer user, password, token;
+    wu->getSecurityToken(StringBufferAdaptor(token));
+    extractToken(token.str(), wu->queryWuid(), StringBufferAdaptor(user), StringBufferAdaptor(password));
```

Comment:
@richardkchapman we have really got to find a better way than storing user credentials in an easily extracted manner within workunits.  Especially if we ever start using credentials that are valid for services outside of HPCC functionality.